x/stake/keeper/delegation.go

```diff
@@ -647,6 +652,7 @@ func (k Keeper) BeginUnbonding(ctx sdk.Context,
 
 	balance := sdk.NewCoin(k.BondDenom(ctx), returnAmount.RawInt())
 
+	delegation, _ := k.GetDelegation(ctx, delAddr, valAddr)
```

Comment:
Can we put `CrossStake` in the `sdk.Context`, so we don't have to query `delegation` again.

Reply:
Done